### PR TITLE
[APM] Rename DD_APM_MAX_TPS config to DD_APM_TARGET_TPS

### PR DESCRIPTION
**Pull request by ichinaski**
### What does this PR do?

Jira Ticket: https://datadoghq.atlassian.net/browse/APMSP-1156

On APM, `DD_APM_MAX_TPS` is an inaccurate name for the setting to control the desired number of traces per second.

This setting is renamed to `DD_APM_TARGET_TPS`, which better reflects the meaning and behavior of this feature.

Changes:

- Deprecate `DD_APM_MAX_TPS` env var and `max_traces_per_second` yaml settings, in favor of `DD_APM_TARGET_TPS` and `target_traces_per_second`, respectively.
- Support both old and new settings, with the new setting taking precedence over the deprecated one.
- Add test cases to ensure backwards compatibility.
- Add E2E test to check the newly updated `DD_APM_TARGET_TPS` is also present in traces sent to the intake.

### CODE REVIEW

File: test/new-e2e/tests/apm/docker_test.go [hunk 1]
```diff
@@ -69,7 +69,7 @@ func TestDockerFakeintakeSuiteUDS(t *testing.T) {
 
 // TestDockerFakeintakeSuiteTCP runs basic Trace Agent tests over the TCP transport
 func TestDockerFakeintakeSuiteTCP(t *testing.T) {
-	options := dockerSuiteOpts(uds, awsdocker.WithAgentOptions(
+	options := dockerSuiteOpts(tcp, awsdocker.WithAgentOptions(
```
Comment (ichinaski) [inline test/new-e2e/tests/apm/docker_test.go:72]: note: this is not related to the PR goals, but a side fix related to the E2E tests that I was working on.

File: releasenotes/notes/apm-rename-target-tps-6e7fce5828fcccb4.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+deprecations:
+  - |
+    APM: DD_APM_MAX_TPS config setting is deprecated in favor of the more
```
Comment (ajgajg1134) [inline releasenotes/notes/apm-rename-target-tps-6e7fce5828fcccb4.yaml:11]: Let's clarify here that the behavior is the same it's just that the name is being changed to more accurately reflect the existing behavior

File: pkg/config/setup/apm.go [hunk 3]
```diff
@@ -90,7 +90,8 @@ func setupAPM(config pkgconfigmodel.Config) {
 	config.BindEnv("apm_config.decoder_timeout", "DD_APM_DECODER_TIMEOUT")
 	config.BindEnv("apm_config.log_file", "DD_APM_LOG_FILE")
 	config.BindEnv("apm_config.max_events_per_second", "DD_APM_MAX_EPS", "DD_MAX_EPS")
-	config.BindEnv("apm_config.max_traces_per_second", "DD_APM_MAX_TPS", "DD_MAX_TPS")
+	config.BindEnv("apm_config.max_traces_per_second", "DD_APM_MAX_TPS", "DD_MAX_TPS") // deprecatedd
```
Comment (ajgajg1134) [inline pkg/config/setup/apm.go:93]:
```suggestion
	config.BindEnv("apm_config.max_traces_per_second", "DD_APM_MAX_TPS", "DD_MAX_TPS") // deprecated
```

**Review by ajgajg1134 — CHANGES_REQUESTED** [review]
Overall this looks good! Just two small changes here then should be good to go!

**Review by buraizu — COMMENTED** [review]
Suggesting an additional minor update for now, but also created DOCS-8155 to follow up after the changes requested above have been made

**Review by ogaca-dd — APPROVED** [review]
LGTM for files owned by ASC

**Review by pierotibou — COMMENTED** [review]
LGTM, thanks for doing this. I'll let Kyle or Andrew step though :)

File: releasenotes/notes/apm-rename-target-tps-6e7fce5828fcccb4.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    accurate DD_APM_TARGET_TPS. Accordingly, when configured via yaml,
```
Comment (buraizu) [inline releasenotes/notes/apm-rename-target-tps-6e7fce5828fcccb4.yaml:12]:
```suggestion
    accurate DD_APM_TARGET_TPS. Accordingly, when configured through YAML,
```

**Review by buraizu — APPROVED** [review]
Thanks, approving with an additional minor suggestion

File: comp/trace/config/setup.go [hunk 5]
```diff
@@ -232,8 +232,12 @@ func applyDatadogConfig(c *config.AgentConfig, core corecompcfg.Component) error
 		c.MaxEPS = core.GetFloat64("apm_config.max_events_per_second")
 	}
 	if core.IsSet("apm_config.max_traces_per_second") {
+		log.Warn("`apm_config.max_traces_per_second` is deprecated, please use `apm_config.target_traces_per_second` instead")
```
Comment (knusbaum) [inline comp/trace/config/setup.go:235]: Is there anything we can link to here or elsewhere that will let users know what's happening?

As a user, I might be confused about how to update my config if I don't know that `max_traces_per_second` and `target_traces_per_second` do the same thing.

Maybe pointing at the release notes or the config template is sufficient?

**Review by knusbaum — APPROVED** [review]
LGTM. One optional comment.

File: comp/trace/config/setup.go (hunk 5 above)
Comment (ichinaski) [reply comp/trace/config/setup.go:235]: As discussed offline, we will leave it as is, not to have to maintain this link (which can potentially be dead). We will sync with TEEs to give a heads up on the change, to anticipate these issues if customers raise them.

**Review by ajgajg1134 — APPROVED** [review]

**Pull request merged**